src/systems/fem_context.C

```diff
@@ -80,8 +80,20 @@ void FEMContext::init_internal_data(const System & sys)
 {
   // Reserve space for the FEAbstract and QBase objects for each
   // element dimension possibility (0,1,2,3)
-  _element_fe.resize(4);
-  _side_fe.resize(4);
+
+  // Below is a workaround for the ICC 19. The original code was:
+  //
+  // _element_fe.resize(4);
+  // _side_fe.resize(4);
```

Comment:
We've seen this error before with the Intel compiler, and chalked it up to incompatibility between the GCC stdlib installed on the machine and the Intel backend.  I tend to believe the code is actually standards-compliant, but am by no means an expert.

 At least according to my reading of  [this](https://en.cppreference.com/w/cpp/container/vector/resize), as of C++11, `vector<T>::resize()` requires that `T` meet the requirements of "MoveInsertable" and "DefaultInsertable" in order to use overload (1). I would think a `map<FEType, unique_ptr<X>>` meets these requirements but admit I'm not an expert on these two concepts. 

The `vector<T>::push_back()` in your version requires *only* [MoveInsertable](https://en.cppreference.com/w/cpp/container/vector/push_back), I assume we are calling overload (2) there, so that may be why the workaround works...

Reply:
That page does seem to explicitly accept the possibility of a throwing move constructor, when referring to the strong exception guarantee:

> In overload (1), if T's move constructor is not noexcept and T is not CopyInsertable into *this, vector will use the throwing move constructor. If it throws, the guarantee is waived and the effects are unspecified.

Reply:
And honestly even the standard line pointed out on StackOverflow has similar wording:

"If an exception is thrown other than by the move constructor of a non-CopyInsertable T there are no effects."

If a non-CopyInsertable T was required to have a noexcept move constructor then that whole "other than" clause would be meaningless.

Reply:
We should still merge this because it's a cheap workaround for a problem with a major compiler, but I'm now leaning towards this being an icpc bug rather than a libmesh bug.

Reply:
My explanation for why the workarounds do work is as follows (but I am no expert either).

The `resize` method fails, because it cannot copy a pair containing an `unique_ptr`. Thus, the tree must have already one storage element allocated. I am assuming, that `map` already creates the root node or something like that (the map is empty), and this root node cannot be copied. Hence, the workaround works because each time a new `map` is created and then moved into the `vector`. In contrast to trying to copy exactly the empty map multiple times into the vector.

I might be wrong, though (and not willing to read more of the STL code, than what I had to read yesterday. ;) ).

Reply:
Agree, one of the commenters also mentioned MSVC 18.00.40629 has the same issue. The SO seems to have generated a lot of interest!